packages/location-management/src/components/LocationTree/index.tsx

```diff
@@ -111,7 +176,35 @@ const Tree: React.FC<TreeProp> = (props: TreeProp) => {
     });
   }
 
-  generateFilterData(data);
+  useEffect(() => {
+    if (!treeData.length) {
+      getBaseTreeNode(accessToken, opensrpBaseURL)
+        .then(async (response) => {
+          dispatch(fetchLocationUnits(response));
+          const hierarchy = await getHierarchy(response, accessToken, opensrpBaseURL);
+          hierarchy.forEach((hier) => {
+            const processed = generateJurisdictionTree(hier);
+            dispatch(fetchAllHierarchies(processed.model));
+          });
+        })
+        .catch(() => sendErrorNotification('An error occurred'));
+    }
+  }, [accessToken, dispatch, opensrpBaseURL, treeData.length]);
+
+  generateFilterData(treeData);
+
+  if (!treeData.length)
+    return (
+      <Spin
+        style={{
```

Comment:
The `Spin` is styled in the client, do we need custom style here?

Reply:
nice catch  🧐🎉